### DISCUSSION TITLE
A way to wrap a mixed mode model for pipelining purposes?

### DISCUSSION THREAD

**Discussion by florisvdf** [danielegrattarola/spektral#109] [Q&A]
Hi,

I'm looking for a way to include a mixed mode GraphConv network in one of my pipelines where I can flexibly pass hyperparameters. My current approach has been to mostly follow the structure of the [MNIST example](https://github.com/danielegrattarola/spektral/blob/master/examples/other/graph_signal_classification_mnist.py), but I have not been able to get it to work. Preferably I would like to wrap the model in a scikit-learn classifier wrapper using the `tf.keras.wrappers.scikit_learn.KerasClassifier`. Is this possible?

I'm suspecting that the use of mixed mode required the custom `train` and `evaluate` functions as seen in the MNIST example to support the sparse filter.  Is there a straightforward workaround?

My apologies for opening another issue in such a short time window.

Cheers

**Comment by danielegrattarola** [comment]
HI, 

I don't have much experience with the Sklearn wrapper but I guess that a possible workaround would be to extend KerasClassifier and re-write the two methods that you mentioned. 
Unfortunately, Keras often assumes that the inputs are dense and with a fairly "strict" `x, y` input format where all tensors have the same first dimension, which should represent the batch size. There isn't much wiggle room outside of that. The TF2-style definition of models lets you work around those restrictions but it breaks a lot of existing features of Keras, like the wrapper.  

I'm not sure if this is the answer you were looking for. 
Cheers, 
Daniele

**Comment by florisvdf** [comment]
Hi Daniele,

Thank you for the elaboration. I was expecting as much. If I manage to come up with a solution, I'll make sure to share it in this thread.

Cheers,
Floris
